packages/frontend/src/pages/new_league.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { useRef, useState } from "react";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faBan, faFloppyDisk, faPlus } from "@fortawesome/free-solid-svg-icons";
+import { Page } from "../components/page";
+import { Header } from "../components/page/header";
+import { useViewNav } from "../state/nav";
+import { Form } from "../components/form/index";
+import { Input } from "../components/form/input";
+import { RadioButton } from "../components/form/radio_button";
+import { SelectSport } from "../components/form/select_sports";
+import { Button } from "../components/button"; // Import Button component
+import { Icon } from "../components/icon";
+import { Sport } from "../../../types/src/utils"; // Update this path
+// import { api } from "../lib/api"; // Assuming this is your API client (commented out)
+
+export function NewLeaguePage() {
+  const viewNavigate = useViewNav();
+  const formRef = useRef(null); // Ref to access Form component instance
+
+  const [selectedOption, setSelectedOption] = useState("public");
+
+  const handleOptionChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSelectedOption(event.target.value);
+  };
+
+  const sportsOptions: Sport[] = [Sport.Badminton, Sport.Tennis, Sport.Squash];
+  const currentSport: Sport = sportsOptions[0];
+
+  const handleCreateLeague = async () => {
+    //I dont know what to do here await api.leagues().new(formData);
+  };
+
+  const handleCancel = () => {
+    viewNavigate("/leagus"); // UPDATE THIS if needed
+  };
+
+  const handleCreateButtonClick = () => {
+    void handleCreateLeague();
+  };
+
+  const [name, setName] = React.useState("");
+  const [startDate, setStartDate] = React.useState("");
+  const [endDate, setEndDate] = React.useState("");
+
+  const handleNameChange = (value: string) => {
+    setName(value);
+  };
+  const handleStartDateChange = (value: string) => {
+    setStartDate(value);
+  };
+
+  const handleEndDateChange = (value: string) => {
+    setEndDate(value);
+  };
+
+  return (
+    <Page>
+      <Page.Header>
+        <Header.Back />
+        New League
+        <Header.Right>
+          {/* Adjust as needed */}
+          <FontAwesomeIcon icon={faPlus} />
+        </Header.Right>
+      </Page.Header>
+
+      <Page.Body>
+        <Form onSubmit={handleCreateLeague}>
+          <div style={{ padding: "10px 0" }} /> {/* Empty div with padding */}
+          <SelectSport sports={sportsOptions} currentSport={currentSport} />
+          <div style={{ padding: "10px 0" }} /> {/* Empty div with padding */}
+          <Input type="text" placeholder="Name" onChange={handleNameChange} />
+          <div style={{ padding: "10px 0" }} /> {/* Empty div with padding */}
```

Comment:
tailwindcss please. Also if you follow the instructions above, you will have the updated components, where the `Input` has a `className` param which you can set through that instead of having the hacky workaround with adding empty divs everywhere